File: 프로그래머스/2/42842. 카펫/카펫.py
# brown: 갈색 격자의 수 
# yellow: 노란색 격자의 수

import math

def solution(brown, yellow):
    for i in range(1, int(math.sqrt(yellow))+1):
        
        if yellow % i == 0:
            y_w = yellow//i
            y_h = i
            b_w = y_w + 2
            b_h = y_h + 2
            
            if b_w*b_h - y_w*y_h == brown:
                return [b_w, b_h]

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;
class Solution {
    // brown : 갈색 격자의 수
    // yellow : 노란색 격자의 수
    // return : 카펫의 가로, 세로 크기를 순서대로 배열에 담아
    public int[] solution(int brown, int yellow) {
        int[] answer = {};

        for(int i=1;i<=Math.sqrt(yellow);i++){
            if (yellow/(double)i % 1 == 0) {
                int w = Math.max(yellow/i, i);
                int h = Math.min(yellow/i, i);
                if(2*w + 2*h + 4 == brown) {
                    return new int[]{w+2, h+2};
                }
            }
        }
        
        return answer;
    }
}